server/methods/addAllUserToRoom.js

```diff
@@ -43,6 +43,12 @@ Meteor.methods({
 				RocketChat.models.Messages.createUserJoinWithRoomIdAndUser(rid, user, {
 					ts: now,
 				});
+				if (room.ro) {
+					Meteor.call('muteUserInRoom', {
```

Comment:
this is rather an expensive method to call for each user and will flood the channel with "user muted" messages.

I'd recommend to change this call to an update on room's object adding all users at the same time to the `muted` field.